scylla-rust-wrapper/src/cluster.rs

```diff
@@ -49,6 +50,11 @@ impl LoadBalancingConfig {
     // as it results in panic due to DefaultPolicyBuilder::build() spawning a tokio task.
     pub(crate) async fn build(self) -> Arc<dyn LoadBalancingPolicy> {
         let mut builder = DefaultPolicyBuilder::new().token_aware(self.token_awareness_enabled);
+        if self.token_awareness_enabled {
+            // Cpp-driver enables shuffling replicas only if token aware routing is enabled.
+            builder =
+                builder.enable_shuffling_replicas(self.token_aware_shuffling_replicas_enabled);
+        }
```

Comment:
I would recommend add option to enable shuffling no matter what:
```suggestion
        shuffling := self.shuffling_replicas_enabled || (self.token_awareness_enabled && self.token_aware_shuffling_replicas_enabled) 
        if self.shuffling {
            // Cpp-driver enables shuffling replicas only if token aware routing is enabled.
            builder =
                builder.enable_shuffling_replicas(shuffling);
        }
```

Reply:
From what I've just checked, this setting in rust-driver applies only for replicas (when token-awareness is enabled) as well. When token-awareness is disabled, we always pick nodes randomly (respecting the preferred dc, rack settings etc.). I think this setting is rather introduced, to disable shuffling for replicas, if user wishes to (since it's enabled by default). This is obviously not recommended, but some users might want to pick replicas deterministically for some reason.